presto-main/src/main/java/io/prestosql/dispatcher/DispatchInfo.java

```diff
@@ -64,19 +80,21 @@ private DispatchInfo(Optional<CoordinatorLocation> coordinatorLocation, Optional
     }
 
     /**
-     * Returns the coordinator's uri.
+     * Returns an object with information about coordinator's URI.
      *
-     * @return an Optional {@link CoordinatorLocation}.
+     * @return an object with information about coordinator's URI
+     * @see java.net.URI
      */
     public Optional<CoordinatorLocation> getCoordinatorLocation()
     {
         return coordinatorLocation;
     }
 
     /**
-     * Returns detailed information about an executed failure, such as its type, message and cause.
+     * Returns detailed information about a failure during query execution, such as its type,
+     * message and cause.
      *
-     * @return an Optional {@link ExecutionFailureInfo}.
+     * @return an object with detailed information about a failure during query execution
```

Comment:
Should maintain the information it returns an Optional

Reply:
added